clarity/contracts/pool/alex-reserve-pool.clar

```diff
@@ -87,7 +95,7 @@
 ;; @returns (response bool)
 (define-public (remove-from-balance (token principal) (amount uint))
   (begin
-    (try! (check-is-approved tx-sender))
+    (asserts! (or (is-ok (check-is-approved)) (is-eq tx-sender (as-contract tx-sender))) ERR-NOT-AUTHORIZED) 
```

Comment:
Nit: can also do:

```clojure
(or (is-ok (check-is-approved)) (is-ok (check-is-contract-owner)))
```

Or create a separate private function for it since the check is repeated in other functions.